### PR TITLE
IOTDB-750 Allow SubDevices in IoTDB.

### PR DESCRIPTION
**Pull request by JulianFeinauer**

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/metadata/mnode/LeafMNode.java [hunk 1]
```diff
@@ -24,14 +24,9 @@
...
 
-import java.util.Collections;
 import java.util.Map;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
-public class LeafMNode extends MNode {
-
-  private static final Logger logger = LoggerFactory.getLogger(LeafMNode.class);
+public class LeafMNode extends InternalMNode {
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/metadata/mnode/LeafMNode.java:29]: Call this MeasurementMNode is better.

Comment (Alima777) [reply server/src/main/java/org/apache/iotdb/db/metadata/mnode/LeafMNode.java:29]: I agree with that. Otherwise the concept of leaf will be confusing.

Comment (JulianFeinauer) [reply server/src/main/java/org/apache/iotdb/db/metadata/mnode/LeafMNode.java]: 👍

**Comment by JulianFeinauer** [comment]
Thanks all reviewers so far, especially the suggestion to change LeafMNode name and to clarify intent (Add Measurements BELOW a Measurement(s), NOT add device below device), so a Node can at the same time be a device and a timeseries. I added all changes and rebased : )

**Pull request merged**